.evergreen/deploy.yml

```
@@ -58,6 +66,26 @@ functions:
           ${PREPARE_SHELL}
           yarn deploy-utils-email
 
+  prepare-user-deploy-vars:
+    - command: shell.exec
+      type: setup
+      params:
+        script: |
+          if ! [ -z "${staging_user_key}" ]; then
+            echo "BUCKET: spruce-${staging_user_key}" >> deploy-vars.yml
+            echo "REACT_APP_USER_KEY: ${staging_user_key}" >> deploy-vars.yml
+            echo "REACT_APP_SPRUCE_URL: https://spruce.${staging_user_key}.evergreen-staging.devprod.mongodb.com" >> deploy-vars.yml
+            echo "REACT_APP_UI_URL: https://evergreen-multi.staging.corp.mongodb.com" >> deploy-vars.yml
+            echo "USER_STAGING_ARN: ${user_staging_arn_prefix}${staging_user_key}.ui_put" >> deploy-vars.yml
+          else
```

I think the step of providing a `staging_user_key` could be automated and just rely on the author field as a default. Would something like this work?
```suggestion
        script: |
          staging_user_key="${staging_user_key}"
          if [ -z "$staging_user_key" ]; then
            echo "No staging_user_key provided. Attempting to determine from author."
            # The staging_user_key is just the author username with a dash instead of a period.
            author=${author}
            staging_user_key=$(echo "$author" | sed 's/\./-/g')
            echo "staging_user_key: '$staging_user_key'"
          fi
          if ! [ -z "$staging_user_key" ]; then
            echo "BUCKET: spruce-$staging_user_key" >> deploy-vars.yml
            echo "REACT_APP_USER_KEY: $staging_user_key" >> deploy-vars.yml
            echo "REACT_APP_SPRUCE_URL: https://spruce.$staging_user_key.evergreen-staging.devprod.mongodb.com" >> deploy-vars.yml
            echo "REACT_APP_UI_URL: https://evergreen-multi.staging.corp.mongodb.com" >> deploy-vars.yml
            echo "USER_STAGING_ARN: ${user_staging_arn_prefix}$staging_user_key.ui_put" >> deploy-vars.yml
          else
```

---

Interested in @ybrill's thoughts here, but my inclination is that we should not make any assumptions about this value in the UI codebase. The value will be passed in by the kernel-tools CLI so users will rarely need to specify it themselves; if they ever do, they could set a [default CLI parameter](https://docs.devprod.prod.corp.mongodb.com/evergreen/Project-Configuration/Parameterized-Builds#command-line-usage).

---

Ahh I missed that the CLI will provide this default value. 

> To force a push to their staging environment, a user can run the task directly with the staging_user_key parameter.

I guess it can still help with this portion.  I would defer to you two on the ideal approach.

---

either way works for me. Though technically the environment could be called anything

---

👍 For the sake of flexibility, let's stick with the param approach.